# Reconciling Multiple Connectivity Scores for Drug Repurposing
Kewalin Samart<sup>1,2,4,\*</sup>, Phoebe Tuyishime<sup>1,3,4,\*</sup>, Arjun Krishnan<sup>4,5,\#</sup>, Janani Ravi<sup>1,\#</sup> <br> <sup>1</sup>Pathobiology and Diagnostic Investigation, Michigan State University; <sup>2</sup>Mathematics, Michigan State University; <sup>3</sup>Food Science and Nutrition, Michigan State University; <sup>4</sup>Computational Mathematics, Science and Engineering; <sup>5</sup>Biochemistry and Molecular Biology, Michigan State University. <br> <sup>\*</sup>These authors contributed equally to this work. <br> <sup>\#</sup>Corresponding authors: [arjun\@msu.edu](mailto:arjun@msu.edu); [janani\@msu.edu](mailto:janani@msu.edu)

## Abstract
The basis of several recent methods for drug repurposing is the key principle that an efficacious drug will reverse the disease molecular 'signature' with minimal side-effects. This principle was defined and popularized by the influential 'connectivity map' study in 2006 regarding reversal relationships between disease- and drug-induced gene expression profiles, quantified by a disease-drug 'connectivity score.' Over the past 15 years, several studies have proposed variations in calculating connectivity scores towards improving accuracy and robustness in light of massive growth in reference drug profiles. However, these variations have been formulated inconsistently using various notations and terminologies even though they are based on a common set of conceptual and statistical ideas. Therefore, we present a systematic reconciliation of multiple disease-drug similarity metrics (ES, css, Sum, Cosine, XSum, XCor, XSpe, XCos, EWCos) and connectivity scores (CS, RGES, NCS, WCS, Tau, CSS, EMUDRA) by defining them using consistent notation and terminology. In addition to providing clarity and deeper insights, this coherent definition of connectivity scores and their relationships provides a unified scheme that newer methods can adopt, enabling the computational drug-development community to compare and investigate different approaches easily. To facilitate the continuous and transparent integration of newer methods, this article will be available as a live document (<https://jravilab.github.io/connectivity_scores>) coupled with a GitHub repository (<https://github.com/jravilab/connectivity_scores>) that any researcher can build on and push changes to.

## Keywords
drug repositioning/repurposing \| disease gene signature \| drug profile \| CMap and LINCS L1000 \| similarity metrics \| connectivity mapping \| transcriptomics

## Key points
- Connectivity mapping is a powerful approach for drug repurposing based on finding drugs that reverse the transcriptional signature of a disease, quantified by a connectivity score.
- Though a number of similarity metrics and connectivity scores have been proposed until now, they have been described using inconsistent notations and terminologies to refer to a common set of concepts and ideas.
- Here, we present a coherent definition of multiple connectivity scores using a unified notation and terminology, along with delineating the clear relationship between these scores.
- Our unified scheme can be adopted easily by newer methods and used for systematic comparisons.
- The live document and GitHub repository will allow continuous incorporation of newer methods.

## Full Text

[arXiv](https://arxiv.org/abs/2009.09317v3) \| [PDF](https://arxiv.org/pdf/2009.09317v3.pdf) \| [HTML Live Doc](https://jravilab.github.io/connectivity_scores)

## About the authors

-   **Kewalin Samart** is an undergraduate junior majoring in Computational Mathematics, and Computational Math, Science & Engineering at Michigan State University. Her research currently focuses on disease and drug expression signatures and host-directed drug repurposing.
-   **Phoebe Tuyishime** is a recent graduate from the College of Agriculture and Natural Resources at Michigan State University. Her research focuses on infectious diseases, computational drug repurposing, and host-directed therapeutics. She will be starting her MS at Univ. British Columbia soon.
-   **Arjun Krishnan** is an Assistant Professor in the Departments of Computational Math, Science & Engineering, and Biochemistry & Molecular Biology at Michigan State University. His group works broadly in the areas of computational genomics and biomedical data science.
-   **Janani Ravi** is an Assistant Professor (Fixed-Term) in Pathobiology and Diagnostic Investigation at Michigan State University. Her research group focuses on protein sequence-structure-function relationships, comparative genomics, and drug repurposing as they relate to infectious diseases.

### Contact

-   Kewalin Samart \| [@KewalinSamart](//twitter.com/KewalinSamart)
-   Phoebe Tuyishime \| [@TuyishimePhoebe](//twitter.com/TuyishimePhoebe)
-   Arjun Krishnan \| [@compbiologist](//twitter.com/compbiologist) \| [arjun\@msu.edu](mailto:arjun@msu.edu)
-   Janani Ravi \| [@janani137](//twitter.com/janani137) \| [janani\@msu.edu](mailto:janani@msu.edu)
